docs/HowToConnectExternally.md

```diff
@@ -0,0 +1,298 @@
+
+### GCP
+
+No additional configuration is needed to enable NuoDB database external access in GKE.
+For more information, check [Configuring TCP/UDP load balancing](https://cloud.google.com/kubernetes-engine/docs/how-to/service-parameters).
+
+### AWS
+
+By default, the NuoDB Helm Charts 3.4.0 excepts that the [AWS Load Balancer Controller](https://docs.aws.amazon.com/eks/latest/userguide/aws-load-balancer-controller.html) is deployed in the Amazon Elastic Kubernetes Service (EKS) and the native CNI plugin is used when provisioning internet-facing load balancer.
+It will automatically provision AWS Network Load Balancer (NLB) for Kubernetes services of type `LoadBalancer`, as described in the [Network load balancing on Amazon EKS](https://docs.aws.amazon.com/eks/latest/userguide/network-load-balancing.html) guide.
+For more information on how to customize the provisioned NLB, check [Network Load Balancer](https://kubernetes-sigs.github.io/aws-load-balancer-controller/latest/guide/service/nlb/).
+When the `internalIP` option is set to “true”, no additional configuration is needed.
+
+If the [legacy AWS cloud provider load balancer controller](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer) is used the Kubernetes service annotations must be changed. For example:
+
+```yaml
+cloud:
+  provider: aws
+
+admin:
+  externalAccess:
+    enabled: true
+    internalIP: false
+    annotations:
+      service.beta.kubernetes.io/aws-load-balancer-scheme: internet-facing
+
+database:
+  te:
+    externalAccess:
+      enabled: true
+      internalIP: false
+      annotations:
+        service.beta.kubernetes.io/aws-load-balancer-scheme: internet-facing
+```
+
+### Azure
+
+No additional configuration is needed to enable NuoDB database external access in Azure Kubernetes Service (AKS).
+For more information, see [Use a public Standard Load Balancer](https://docs.microsoft.com/en-us/azure/aks/load-balancer-standard).
+
+## Troubleshooting
+
+### Unable to connect
+
+There may be different reasons for client connectivity problems such as:
+
+- not _Ready_ NuoDB Admin Pods
+- not _Ready_ TE pods
+- incorrect external access configuration
+- incorrect NuoDB Load Balancer configuration
+- incorrect NLB configuration
+- network connectivity problems including lack of routing, firewall configuration, and many more
+
+> **ACTION**: You can rule out any of the points above one by one.
+Start by checking the overall Pod status for the NuoDB domain and database.
+Some of the common troubleshooting steps are listed below, however, there might be additional verifications specific to your deployment.
+
+1. Verify that all AP, TE, and SM pods are reported _Ready_ using `kubectl get pods`.
+2. Check the NuoDB domain and database using `nuocmd show domain`.
+3. Verify the database availability inside the cluster is using the same connection properties as the application uses.
```

Comment:
Should this say something like "Verify that the database is available to SQL clients within the Kubernetes cluster by using the same connection properties that the application uses"?

Reply:
Your suggestion read better. Will change the text.